gems/sorbet-runtime/test/types/attached_class.rb

```diff
@@ -20,6 +20,20 @@ class A < Base; end
       assert_equal(A.make.class, A)
     end
 
+    it 'can type self methods that use self.new' do
+
+      class Base
+        extend T::Sig
+
+        sig {returns(T.experimental_attached_class)}
+        def self.make
+          self.new
+        end
+      end
+
+      assert_equal(Base.make.class, Base)
+    end
+
```

Comment:
Might be worth instantiating a child of base here too.

Reply:
Good idea.